src/common/actions/repository.js

```diff
@@ -22,11 +23,18 @@ export const REPO_RESET = 'REPO_RESET';
 export function addRepos(repos, owner) {
   return async (dispatch) => {
     try {
-      await Promise.all(repos.map(async (repo) => {
+      await Promise.all(repos.map( async (repo) => {
         await dispatch(createWebhook(repo));
         return dispatch(addRepo(repo));
       }));
 
+      // XXX
+      // Wait for updated snaps list (so we don't show My repos until we
+      // updated the list with newly added repos).
+      //
+      // This is a temporary fix before we implement #666
+      await dispatch(fetchUserSnaps(owner));
```

Comment:
`addRepos` doesn't need to fetch user snaps, but thanks to that we will not redirect back to "My repos" until snaps list is updated, so when we do redirect back the list will already be updated with newly added snaps.

It adds additional request we need to wait for, so it would be best to progressively load (as described in #666), but for now it gives better user experience by not flashing old (or empty) My repos before data is loaded.